source/frontend/src/models/api/Project.ts

```diff
@@ -1,31 +1,58 @@
 import { Api_AcquisitionFile } from './AcquisitionFile';
 import { Api_AuditFields } from './AuditFields';
 import { Api_CodeType } from './CodeType';
-import { Api_ConcurrentVersion } from './ConcurrentVersion';
+import { Api_ConcurrentVersion_Null } from './ConcurrentVersion';
 import Api_TypeCode from './TypeCode';
 
-export interface Api_Project extends Api_ConcurrentVersion, Api_AuditFields {
-  id?: number;
-  projectStatusTypeCode?: Api_TypeCode<string>;
-  businessFunctionCode?: Api_CodeType;
-  costTypeCode?: Api_CodeType;
-  workActivityCode?: Api_CodeType;
-  regionCode?: Api_CodeType;
-  code?: string;
-  description?: string;
-  note?: string;
-  products?: Api_Product[];
+export interface Api_Project extends Api_ConcurrentVersion_Null, Api_AuditFields {
+  id: number | null | undefined;
+  projectStatusTypeCode: Api_TypeCode<string> | null;
+  businessFunctionCode: Api_CodeType | null;
+  costTypeCode: Api_CodeType | null;
+  workActivityCode: Api_CodeType | null;
+  regionCode: Api_CodeType | null;
+  code: string | null;
+  description: string | null;
+  note: string | null;
+  products: Api_Product[];
 }
 
-export interface Api_Product extends Api_ConcurrentVersion, Api_AuditFields {
-  id?: number;
-  parentProject?: Api_Project;
-  code?: string;
-  description?: string;
-  startDate?: string;
-  costEstimate?: number;
-  costEstimateDate?: string;
-  objective?: string;
-  scope?: string;
-  acquisitionFiles?: Api_AcquisitionFile[];
+export const defaultProject: Api_Project = {
+  id: null,
+  projectStatusTypeCode: null,
+  businessFunctionCode: null,
+  workActivityCode: null,
+  code: null,
+  costTypeCode: null,
+  regionCode: null,
+  description: null,
+  note: null,
+  products: [],
+  rowVersion: null,
+};
+
+export interface Api_Product extends Api_ConcurrentVersion_Null, Api_AuditFields {
+  id?: number | null | undefined;
```

Comment:
Q: Why is it necessary that the IDs support undefined?

Reply:
It doesn't need to explicitly, I'll change this.